homeassistant/components/tplink/__init__.py

```diff
@@ -0,0 +1,62 @@
+"""Component to embed TP-Link smart home devices."""
+import logging
+
+from homeassistant import config_entries
+from homeassistant.helpers import config_entry_flow
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'tplink'
+REQUIREMENTS = ['pyHS100==0.3.2']
+
+
+async def async_setup(hass, config):
+    """Set up the TP-Link component."""
+    conf = config.get(DOMAIN)
+
+    hass.data[DOMAIN] = conf or {}
+
+    if conf is not None:
+        hass.async_create_task(hass.config_entries.flow.async_init(
+            DOMAIN, context={'source': config_entries.SOURCE_IMPORT}))
+
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up Sonos from a config entry."""
+    from pyHS100 import Discover, SmartBulb, SmartPlug
+
+    hass.data[DOMAIN] = {'light': [], 'switch': []}
```

Comment:
What if user has manual configured devices already imported, will this overwrite it?

Reply:
The previous mechanism does not use any part of this, so it has no idea about auto-detected devices, but the ones which came after would error out due to a clash with unique ids.

What would be the best way to handle this? I'm unsure if this is going to be initialized always and what's the initialization order. And if this is not initialized always, where should the dictionary be initialized? I could adjust the `setup_platform()`s to check for an existing device and bail out if one already exists, but the initialization should happen somewhere else. 

This all assumes that the initialization order is 1) manually configured devices from configuration.yaml etc. and first after that 2) auto-detected devices. Is this a correct assumption?

Reply:
I think we should respect user configured device, ignore duplicated finding in discovery. Our users are so smart, I had seen complaints about cannot discovery Sonos or Cast (since they put them in separate VLAN etc), and they are asking keep the capability to add device through configuraiton.yaml.

Reply:
Yes, an ability to configure these locally is a must have of course. The information stored in `hass.data` is used only for auto-detected devices at the moment (it stores already initialized instances), so I think the proper way to fix this is:
1. Add manually configured devices to those lists.
2. Check the list for unique id to before initializing the discovered devices.

The remaining question is where should that dictionary be initialized?

Reply:
Things should be initialized as close to where it is going to be used. So the `switch.tplink` platform will only use it, it should live there.

Reply:
Oh I see now that we do discovery once in the component, in that case initialize it here.